app/v1/services/EnrolmentsAuthService.scala

```diff
@@ -0,0 +1,131 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v1.services
+
+import javax.inject.{Inject, Singleton}
+import play.api.Logger
+import play.api.libs.json.JsResultException
+import uk.gov.hmrc.auth.core.AffinityGroup.{Agent, Individual, Organisation}
+import uk.gov.hmrc.auth.core._
+import uk.gov.hmrc.auth.core.authorise.Predicate
+import uk.gov.hmrc.auth.core.retrieve.v2.Retrievals
+import uk.gov.hmrc.auth.core.retrieve.v2.Retrievals._
+import uk.gov.hmrc.auth.core.retrieve.~
+import uk.gov.hmrc.http.HeaderCarrier
+import v1.models.auth.UserDetails
+import v1.models.errors.{DownstreamError, ForbiddenDownstreamError, LegacyUnauthorisedError}
+import v1.models.outcomes.AuthOutcome
+
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class EnrolmentsAuthService @Inject()(val connector: AuthConnector) {
+
+  private val authFunction: AuthorisedFunctions = new AuthorisedFunctions {
+    override def authConnector: AuthConnector = connector
+  }
+
+  def authorised(predicate: Predicate)(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[AuthOutcome] = {
+    authFunction.authorised(predicate).retrieve(affinityGroup and allEnrolments) {
+      case Some(Individual) ~ _ =>
+        val user: UserDetails =
+          UserDetails(
+            userType = "Individual",
+            agentReferenceNumber = None,
+            clientId = ""
+          )
+
+        successLogging(Individual)
+        Future.successful(Right(user))
+      case Some(Organisation) ~ _ =>
+        val user: UserDetails =
+          UserDetails(
+            userType = "Organisation",
+            agentReferenceNumber = None,
+            clientId = ""
+          )
+
+        successLogging(Organisation)
+        Future.successful(Right(user))
+      case Some(Agent) ~ _ =>
+        val user: UserDetails =
+          UserDetails(
+            userType = "Agent",
+            agentReferenceNumber = None,
+            clientId = ""
+          )
+
+        successLogging(Agent)
+        retrieveAgentReference().map(arn => Right(user.copy(agentReferenceNumber = arn)))
+      case _ =>
+        Logger.error(s"[AuthorisationService] [authoriseAsClient] Authorisation failed due to unsupported affinity group.")
+        Future.successful(Left(LegacyUnauthorisedError))
+    } recoverWith unauthorisedError
+  }
+
+  private def successLogging(affinityGroup: AffinityGroup)(implicit hc: HeaderCarrier, ec: ExecutionContext): Unit = {
+
+    val clientReference = retrieveClientReference(affinityGroup).map {
+      case Some(clientReference) => clientReference
+      case _ => ""
+    }
+
+    Logger.debug(s"[AuthorisationService] [authoriseAsClient] Authorisation succeeded as fully-authorised organisation " +
+      s"for VRN $clientReference.")
+  }
+
+  private def retrieveClientReference(affinityGroup: AffinityGroup)(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[Option[String]] = {
```

Comment:
what is the purpose of this method?? and each time we call this method we are making auth call?